WordPress/gutenberg
# Font family: switch from CSS Custom Property to classes

Fixes #31489

This PR switches the font-family property to use class names instead of CSS Custom Properties, following what the other properties do.

## How has this been tested?

- Use an FSE-enabled theme (e.g. TT1 Blocks).
- Create a new post, and insert a block that has support for the font-family style enabled (e.g. Verse).
- In the block sidebar's typography panel, change the font family.
- Verify that the block's font changes accordingly.
- Check the Code Editor. Verify that it matches the below snippet.
- Exit the Code Editor.
- Publish the post, and view it in the front end. Verify that the font family class name is applied to the frontend.

## Block markup changes

### Before

```html

<pre class="wp-block-verse" style="font-family:var(--wp--preset--font-family--geneva-verdana)">Font family</pre>

```

### After

```html

<pre class="wp-block-verse has-geneva-verdana-font-family">Font family</pre>

```

## Screencasts

### Before

![font-family-before](https://user-images.githubusercontent.com/96308/118710163-37d27080-b81e-11eb-889b-aac3a1ba3791.gif)

### After

![font-family-after](https://user-images.githubusercontent.com/96308/118710242-52a4e500-b81e-11eb-964d-4bf09112db0d.gif)

## Tasks

- [x] Update server hook.
- [x] Update client hook.
- [x] Tell the engine to generate the font-family CSS Custom Properties and classes.
- [x] Add deprecations to move the font-family information from `style.typography.fontFamily` into a top-level `fontFamily` attribute.
  - [x] Navigation
  - [x] Post Comment Author
  - [x] Post Comment Date
  - [x] Post Date
  - [x] Post title
  - [x] Query title
  - [x] Site tagline
  - [x] Site title
  - [x] Verse
